RuriLib/Models/Blocks/Custom/ScriptBlockInstance.cs

```diff
@@ -175,23 +176,22 @@ public override string ToCSharp(List<string> definedVariables, ConfigSettings se
 
                 case Interpreter.NodeJS:
                     var nodeScript = @$"module.exports = async ({MakeInputs()}) => {{
-{Script}
-var noderesult = {{
-{MakeNodeObject()}
-}};
-return noderesult;
-}}";
+                        {Script}
+                        var noderesult = {{
+                        {MakeNodeObject()}
+                        }};
+                        return noderesult;
+                        }}";
 
                     scriptHash = HexConverter.ToHexString(Crypto.MD5(Encoding.UTF8.GetBytes(nodeScript)));
+
                     scriptPath = $"Scripts/{scriptHash}.{GetScriptFileExtension(Interpreter)}";
 
-                    if (!Directory.Exists("Scripts"))
-                        Directory.CreateDirectory("Scripts");
+                    string escapedScript = JsonConvert.ToString(nodeScript);
 
-                    if (!File.Exists(scriptPath))
-                        File.WriteAllText(scriptPath, nodeScript);
+                    (string, string) scriptNHash = (escapedScript, JsonConvert.ToString(scriptPath));
```

Comment:
Why is `scriptPath` being used as a hash here? `scriptHash` is enough. There is no need to use a file anymore, the file-based invocation will only stay in the `InvokeNode` method for backwards compatibility, so `scriptPath` can be removed. Also, since it's just a hex string, it does not need to be escaped in this case, and can be passed to the writer like

```cs
writer.WriteLine($"var {resultName} = await InvokeNode<dynamic>(data, {escapedScript}, new object[] {{ {InputVariables} }}, true, \"{scriptHash}\");");
```